src/main/java/com/cisco/mongodb/aggregate/support/query/AbstractAggregateQueryProvider.java

```diff
@@ -131,6 +146,17 @@ protected String deriveCollectionName(Class className) {
     return collectionName;
   }
 
+  private Expression detectExpression(String collectionName) {
+    Expression expression = null;
+    try {
+      expression = EXPRESSION_PARSER.parseExpression(collectionName, ParserContext.TEMPLATE_EXPRESSION);
+    }
+    catch (ParseException pe) {
+      LOGGER.warn("Not able to parse the collection name expression {}", collectionName);
```

Comment:
throw IllegalArgumentException here instead.  No point continuing on with an unparseable expression and failing further downstream